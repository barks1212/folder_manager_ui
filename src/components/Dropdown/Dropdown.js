import React from 'react';
import PropTypes from 'prop-types';

import {noop} from '../../utils/utils';

const dropdown = ({ setSort }) => (
    <select defaultValue="Sort By" onChange={(e) => setSort(e.target.value)}>
        <option value="Sort By" disabled>Sort by</option>
        <option value="name">Name</option>
        <option value="added">Date</option>
    </select>
);

dropdown.propTypes = {
    setSort: PropTypes.func
};

dropdown.defaultProps = {
    setSort: noop
};

export default dropdown;