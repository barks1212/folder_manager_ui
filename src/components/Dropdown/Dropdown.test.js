import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react'

import Dropdown from './Dropdown';

afterEach(cleanup);

describe('dropdown', () => {

    it('should render', () => {
        const setSortMock = jest.fn();

        const {asFragment} = render(<Dropdown setSort={setSortMock}/>)
        expect(asFragment()).toMatchSnapshot();
    });

    it('should call set sort on change value', () => {
        const setSortMock = jest.fn();
        const {container} = render(<Dropdown setSort={setSortMock} />)

        const dropdown = container.querySelector('select');
        fireEvent.change(dropdown, {target: {value:'added'}});

        expect(setSortMock).toHaveBeenCalled();
    });
});