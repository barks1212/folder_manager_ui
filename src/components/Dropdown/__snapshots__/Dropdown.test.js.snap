// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown should render 1`] = `
<DocumentFragment>
  <select>
    <option
      disabled=""
      selected=""
      value="Sort By"
    >
      Sort by
    </option>
    <option
      value="name"
    >
      Name
    </option>
    <option
      value="added"
    >
      Date
    </option>
  </select>
</DocumentFragment>
`;
